src/spec-common/injectHeadless.ts

```diff
@@ -484,38 +484,49 @@ async function runLifecycleCommand({ lifecycleHook }: ResolverParameters, contai
 			},
 			onDidChangeDimensions: lifecycleHook.output.onDidChangeDimensions,
 		}, LogLevel.Info);
-		try {
-			const remoteCwd = containerProperties.remoteWorkspaceFolder || containerProperties.homeFolder;
-			async function runSingleCommand(postCommand: string | string[], name?: string) {
-				const progressDetail = typeof postCommand === 'string' ? postCommand : postCommand.join(' ');
-				infoOutput.event({
-					type: 'progress',
-					name: progressName,
-					status: 'running',
-					stepDetail: progressDetail
-				});
+		const remoteCwd = containerProperties.remoteWorkspaceFolder || containerProperties.homeFolder;
+		async function runSingleCommand(postCommand: string | string[], name?: string) {
+			const progressDetail = typeof postCommand === 'string' ? postCommand : postCommand.join(' ');
+			infoOutput.event({
+				type: 'progress',
+				name: progressName,
+				status: 'running',
+				stepDetail: progressDetail
+			});
 
-				// If we have a command name then the command is running in parallel and 
-				// we need to hold output until the command is done so that the output
-				// doesn't get interleaved with the output of other commands.
-				const printMode = name ? 'off' : 'continuous';
-				const env = { ...(await remoteEnv), ...(await secrets) };
+			// If we have a command name then the command is running in parallel and 
+			// we need to hold output until the command is done so that the output
+			// doesn't get interleaved with the output of other commands.
+			const printMode = name ? 'off' : 'continuous';
+			const env = { ...(await remoteEnv), ...(await secrets) };
+			try {
 				const { cmdOutput } = await runRemoteCommand({ ...lifecycleHook, output: infoOutput }, containerProperties, typeof postCommand === 'string' ? ['/bin/sh', '-c', postCommand] : postCommand, remoteCwd, { remoteEnv: env, pty: true, print: printMode });
 
 				// 'name' is set when parallel execution syntax is used.
 				if (name) {
 					infoOutput.raw(`\x1b[1mRunning ${name} from ${userCommandOrigin}...\x1b[0m\r\n${cmdOutput}\r\n`);
 				}
-
-				infoOutput.event({
-					type: 'progress',
-					name: progressName,
-					status: 'succeeded',
-				});
+			} catch (err) {
+				if (err?.cmdOutput) {
```

Comment:
This is still missing an additional check. Maybe:
```suggestion
				if (printMode === 'off' && err?.cmdOutput) {
```